application/helpers/remotecontrol/remotecontrol_handle.php

```diff
@@ -474,45 +474,70 @@ public function set_survey_properties($sSessionKey, $iSurveyID, $aSurveyData)
-
-                if (isset($aActivateResults['error'])) {
-                    return array('status' => 'Error: ' . $aActivateResults['error']);
-                } else {
-                    return $aActivateResults;
-                }
-            } else {
-                            return array('status' => 'No permission');
-            }
-        } else {
-                    return array('status' => self::INVALID_SESSION_KEY);
+        // check provided session key and setup user session
+        if (!$this->_checkSessionKey($sSessionKey)) {
+            return array('status' => self::INVALID_SESSION_KEY);
+        }
+        $iSurveyID = (int) $iSurveyID;
+        // check valid survey id and current user has permission
+        if (!Permission::model()->hasSurveyPermission($iSurveyID, 'surveyactivation', 'update')) {
+            return array('status' => 'No permission');
         }
+        $oSurvey = Survey::model()->findByPk($iSurveyID);
+        // check if survey is already activated
+        if ($oSurvey->isActive) {
+            return array('status' => 'Error: Survey already active');
+        }
+        // Check consistency for groups and questions
+        Yii::app()->loadHelper('admin/activate');
+        $checkHasGroup = checkHasGroup($iSurveyID);
+        $checkGroup = checkGroup($iSurveyID);
+        if ($checkHasGroup !== false || $checkGroup !== false) {
+            return array('status' => 'Error: Survey does not pass consistency check');
+        }
+        $surveyActivator = new SurveyActivator($oSurvey);
+        // grab computed survey settings from common helper
+        $surveysettings = getSurveyInfo($iSurveyID);
+        // list of activation related survey settings and allowed values
+        $activationSettingNames = [
+            'anonymized',
+            'datestamp',
+            'ipaddr',
+            'ipanonymize',
+            'refurl',
+            'savetimings',
+        ];
+        $allowedSettingValues = ['Y', 'N'];
+        // change survey activation settings based on survey settings
+        // and inheritance from survey group and/or global survey
+        foreach ($activationSettingNames as $activationSettingName) {
+            $oSurvey->{$activationSettingName} = $surveysettings[$activationSettingName];
+        }
+        // override computed activation settings with values passed in by user
+        foreach ($activationSettingNames as $activationSettingName) {
+            if (array_key_exists($activationSettingName, $activationSettings)) {
+                $activationSettingValue = $activationSettings[$activationSettingName];
+                if (in_array($activationSettingValue, $allowedSettingValues)) {
+                    $oSurvey->{$activationSettingName} = $activationSettingValue;
+                }
+            }
+        }
+        // persist changes
+        $oSurvey->save();
+        // Make sure the saved values will be picked up
+        Survey::model()->resetCache();
```

Comment:
Not needed : $oSurvey->anonymized is already set in current state here. And we don't put result in cache : we only put table definition.

Reply:
> Not needed : $oSurvey->anonymized is already set in current state here. And we don't put result in cache : we only put table definition.

This is from https://github.com/LimeSurvey/LimeSurvey/blob/ea1c4ba3b12dd817bc820c6239b72e4d476ccd7d/application/controllers/SurveyAdministrationController.php#L1870

I worked from the existing functionality from the admin activate button/dialog

Reply:
seems to work without it though, I can remove